desktop_app/src/backend/server/plugins/mcpServer/index.ts

```diff
@@ -47,6 +79,31 @@ const mcpServerRoutes: FastifyPluginAsyncZod = async (fastify) => {
     }
   );
 
+  fastify.get(
+    '/api/mcp_server/catalog/search',
+    {
+      schema: {
+        operationId: 'searchMcpServerCatalog',
+        description: 'Search for MCP servers in the catalog',
+        tags: ['MCP Server'],
+        querystring: CatalogSearchParamsSchema,
+        response: {
+          200: CatalogSearchResponseSchema,
+          500: ErrorResponseSchema,
+        },
+      },
+    },
+    async ({ query }, reply) => {
+      try {
+        const response = await McpServerModel.searchCatalog(query);
+        return reply.send(response);
+      } catch (error: any) {
+        log.error('Failed to search MCP server catalog:', error);
+        return reply.code(500).send({ error: 'Failed to search catalog' });
+      }
+    }
+  );
```

Comment:
I believe this isn't needed/used?

the UI already has a codegen'd catalog client which is used [here](https://github.com/archestra-ai/archestra/blob/main/desktop_app/src/ui/stores/connector-catalog-store.ts#L82)